codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -548,7 +548,14 @@ public static String formatHostnameForHttp(InetSocketAddress addr) {
             if (!addr.isUnresolved()) {
                 hostString = NetUtil.toAddressString(addr.getAddress());
             }
-            return '[' + hostString + ']';
+
+            // If IPv6 address already contains brackets, let's return as is.
+            if (hostString.charAt(0) == '[' && hostString.charAt(hostString.length() - 1) == ']') {
+                return hostString;
```

Comment:
@hyperxpro shouldn't we only do this if its `unresolved` ?

Reply:
Right, yes.